pkg/kubelet/config/common.go

```diff
@@ -144,19 +149,11 @@ func tryDecodeSinglePod(data []byte, defaultFn defaultFunc) (parsed bool, pod *v
 		return true, nil, err
 	}
 
-	for _, v := range v1Pod.Spec.Volumes {
-		if v.Projected == nil {
-			continue
+	if utilfeature.DefaultFeatureGate.Enabled(features.PreventStaticPodAPIReferences) {
+		// Check if pod has references to API objects
+		if _, err = podutil.HasAPIObjectReferences(newPod); err != nil {
+			return true, nil, err
```

Comment:
wrap the returned resource in an error here, e.g.: `fmt.Errorf("static pods may not reference %s API objects")`